### DISCUSSION TITLE
ARM machines cannot be selected by KKP

### DISCUSSION THREAD

**Discussion by clickersmudge** [kubermatic/kubermatic#12673] [Q&A]
Hi,
We have installed the latest version of KKP (kubermatic-ce-v2.23.4-linux-amd64) on Hetzner servers.

In the dashboard we cannot select ARM machines or an operating system, e.g. Ubuntu ARM.

![Zrzut ekranu z 2023-09-29 12-37-32](https://github.com/kubermatic/kubeone/assets/113194014/67921479-5eaa-4119-a463-e18dc223d313)

There are only "Dedicated" and "Standard" machines.

Do we need to add anything additional to the seed to be able to select them?
It seems to us that once we choose an ARM machine, we will also need to choose a compatible operating system.
Is it currently possible to do this from the panel?

**Comment by xrstf** [comment]
Hi there,

it sounds like this discussion would fit better into the KKP discussions (https://github.com/kubermatic/kubermatic/discussions), as I cannot see what KubeOne has to do with the issue :)

Regardless, you are right in that the dashboard only allows certain machines: https://github.com/kubermatic/dashboard/blob/main/modules/api/pkg/handler/common/provider/hetzner.go#L113-L118 -- only machine types `CX..` and `CCX..` are currently in the allowlist (for comparison, ARM machines seem to be using `CAX..` prefixes). That code was originally written in kubermatic/kubermatic#3952, long before ARM machines were a thing on the Hetzner Cloud. I don't think there's a specific reason to exclude ARM machines nowadays in KKP.

This should be easy to fix however by just extending the `switch` statement a bit. Anyone up for the task?

**Reply by clickersmudge** [reply]
@xrstf You are right, I made a mistake. I meant to create this thread in KKP. I will create an issue in KKP. Thank you for your help.

**Reply by xrstf** [reply]
Thanks :)

**Reply by kron4eg** [reply]
@clickersmudge I've moved the discussion to kubermatic repo.
